pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -1039,25 +1039,40 @@ public void addUnAckedMessages(int numberOfMessages) {
 
     @Override
     public void afterAckMessages(Throwable exOfDeletion, Object ctxOfDeletion) {
-        if (blockedDispatcherOnCursorDataCanNotFullyPersist == TRUE) {
-            if (cursor.isCursorDataFullyPersistable()) {
-                // If there was no previous pause due to cursor data is too large to persist, we don't need to manually
-                // trigger a new read. This can avoid too many CPU circles.
-                if (BLOCKED_DISPATCHER_ON_CURSOR_DATA_CAN_NOT_FULLY_PERSIST_UPDATER.compareAndSet(this, TRUE, FALSE)) {
-                    readMoreEntriesAsync();
-                } else {
-                    // Retry due to conflict update.
-                    afterAckMessages(exOfDeletion, ctxOfDeletion);
-                }
+        boolean paused = blockedDispatcherOnCursorDataCanNotFullyPersist == TRUE;
+        // Trigger a new read if needed.
+        boolean shouldPauseNow = checkAndResumeIfPaused();
```

Comment:
`checkAndResumeIfPaused` means the dispatcher is paused before, so we decide to change it not to pause.
So `shouldNotPause` or `shouldResume`

Reply:
Changed the return value to `shouldResume`